File: Leetcode/1394. Find Lucky Integer in an Array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        map<int,int> m;
        for(auto x:arr)m[x]++;
        int maxn = -1;
        for(auto y:m){
            if(y.first==y.second)maxn = max(maxn,y.first);
        }
        return maxn;
    }
};

File: Leetcode/1394. Find Lucky Integer in an Array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        arr_dict = {}
        for i in arr:
            arr_dict[i] = arr_dict.get(i, 0) + 1
        lucky_str = []
        for key,value in arr_dict.items():
            if key == value:
                lucky_str.append(key)
        
        if len(lucky_str)==0:
            return -1
        else:
            return max(lucky_str)